tests/unit_tests/ensemble_evaluator/test_ensemble_evaluator.py

```diff
@@ -18,11 +12,114 @@
+        events = monitor.track()
+        # first snapshot before any event occurs
+        snapshot_event = next(events)
+        snapshot = Snapshot(snapshot_event.data)
+        assert snapshot.status == ENSEMBLE_STATE_UNKNOWN
+        # two dispatchers connect
+        with Client(
+            url + "/dispatch",
+            cert=cert,
+            token=token,
+            max_retries=1,
+            timeout_multiplier=1,
+        ) as dispatch1, Client(
+            url + "/dispatch",
+            cert=cert,
+            token=token,
+            max_retries=1,
+            timeout_multiplier=1,
+        ) as dispatch2:
+            # first dispatcher informs that job 0 is running
+            send_dispatch_event(
+                dispatch1,
+                identifiers.EVTYPE_FM_JOB_RUNNING,
+                f"/ert/ensemble/{evaluator.ensemble.id_}/real/0/step/0/job/0",
+                "event1",
+                {"current_memory_usage": 1000},
+            )
+
+            # second dispatcher informs that job 0 is running
+            send_dispatch_event(
+                dispatch2,
+                identifiers.EVTYPE_FM_JOB_RUNNING,
+                f"/ert/ensemble/{evaluator.ensemble.id_}/real/1/step/0/job/0",
+                "event1",
+                {"current_memory_usage": 1000},
+            )
+            send_dispatch_event(
+                dispatch2,
+                identifiers.EVTYPE_FM_JOB_RUNNING,
+                f"/ert/ensemble/{evaluator.ensemble.id_}/real/1/step/0/job/1",
+                "event1",
+                {"current_memory_usage": 1000},
+            )
+
+        evt = next(events)
+        snapshot = Snapshot(evt.data)
+        assert snapshot.get_job("0", "0", "0").status == JOB_STATE_RUNNING
+        assert snapshot.get_job("1", "0", "0").status == JOB_STATE_RUNNING
+        assert snapshot.get_job("1", "0", "1").status == JOB_STATE_RUNNING
+        # take down first monitor by leaving context
+
+    with Client(
+        url + "/dispatch",
+        cert=cert,
+        token=token,
+        max_retries=1,
+        timeout_multiplier=1,
+    ) as dispatch2:
+        # second dispatcher informs that job 0 is done
+        send_dispatch_event(
+            dispatch2,
+            identifiers.EVTYPE_FM_JOB_SUCCESS,
+            f"/ert/ensemble/{evaluator.ensemble.id_}/real/1/step/0/job/0",
+            "event1",
+            {"current_memory_usage": 1000},
+        )
+
+        # second dispatcher informs that job 1 is failed
+        send_dispatch_event(
+            dispatch2,
+            identifiers.EVTYPE_FM_JOB_FAILURE,
+            f"/ert/ensemble/{evaluator.ensemble.id_}/real/1/step/0/job/1",
+            "event_job_1_fail",
+            {identifiers.ERROR_MSG: "error"},
+        )
+
+    # we have to wait for the dispatcher to process the events, and for the internal
+    # ensemble state to get updated before connecting and getting a full ensemble
+    # snapshot
+    time.sleep(2)
```

Comment:
Now I'm confused a bit. From the context above I understood that dispatcher is essentially job_runner.Client, which doesn't process anything. But from this one it seems that you refer to BatchingDispatcher. Maybe I don't see inside so well :smile: